contracts/Smt.metasol

```diff
@@ -156,6 +165,69 @@ library SMT{{hashname}} {
     function inc(uint8 x) pure private returns (uint8) {
         unchecked { return x + 1; }
     }
+
+    function modifyElement(MerkleTree storage tree, uint index, bytes memory data) internal {
+        require(tree.elementData[index].length != 0, "Element doesn't exists");
+        _modifyElement(tree, index, data);
+    }
+
+    {{ if SET }}
+    function addToNextEmpty(MerkleTree storage tree, bytes memory data) internal returns (uint256) {
+        uint256 index = tree.firstEmptySlot;
+        tree.firstEmptySlot += 1;
+
+        require(index < (1 << tree.maxDepth), "Index out of range");
+        if (index == (1 << tree.depth)) {
+            _increaseDepth(tree, 1);
+        }
+
+        _addElement(tree, index, data);
+        return index;
+    }
+
+    function removeAndRebase(MerkleTree storage tree, uint index) internal returns (bool){
```

Comment:
why `bool`?

Reply:
if true - that's mean that we write last element to index and remove last element
if false - index it's last element

Reply:
How do we use this information?

Reply:
we will need it in our smart contract when we "unstake"